In [1]:
import numpy as np
import math

In [4]:
#Calculating Limits Numerically, a
def f(x):
    return 2*x**3 - 4*x + 1

# Values approaching 3 from the left and right
x_values = [2.9, 2.99, 2.999, 3.001, 3.01, 3.1]

print("x\t\tf(x)")
for x in x_values:
    print(f"{x}\t{f(x)}")

x		f(x)
2.9	38.178
2.99	42.50179800000001
2.999	42.95001799800001
3.001	43.050018001999995
3.01	43.50180199999999
3.1	48.18200000000001


In [5]:
#Calculate when x is at 0
def g(x):
    return (math.exp(x) - 1) / x

x_values = [-0.1, -0.01, -0.001, 0.001, 0.01, 0.1]

print("x\t\tg(x)")
for x in x_values:
    print(f"{x}\t{g(x)}")

x		g(x)
-0.1	0.9516258196404048
-0.01	0.9950166250831893
-0.001	0.9995001666249781
0.001	1.0005001667083846
0.01	1.005016708416795
0.1	1.0517091807564771


#answer for a: The average is 43.05, so when searching for lim f(x) when x is 3, 43 would be the value. 

#answer for b: When searching for lim f(x), when x is at 0, you can see that the average is approaching 1. So the value would round up to 1. 
#answer for c: For problem a, yes limits can be calculated by plugging in the value where the limit is taken into the function since it's a polynomial. For problem b, no it would not be possible, as plugging in 0 would make the function undefined. A would be continous everywhere since its polynomial function allows for this. For B, this would mean even though the limit exists at x=0 and equals to 1, the line is discontinous with x = 0.  

In [8]:
#Average Rate of Change to Instantaneous Rate of Change
def average_rate_of_change(function, a, b):
    return (function(b) - function(a)) / (b - a)

In [10]:
def distance(t):
    return 4.9 * t**2

In [12]:
#a
avg1 = average_rate_of_change(distance, 5, 6)

print("Average speed from 5 to 6 seconds:", avg1)

Average speed from 5 to 6 seconds: 53.89999999999999


In [13]:
#b
avg2 = average_rate_of_change(distance, 5, 5.5)

print("Average speed from 5 to 5.5 seconds:", avg2)

Average speed from 5 to 5.5 seconds: 51.45000000000002


In [14]:
#c
avg3 = average_rate_of_change(distance, 5, 5.1)

print("Average speed from 5 to 5.1 seconds:", avg3)

Average speed from 5 to 5.1 seconds: 49.490000000000016


In [15]:
#d
intervals = [5.5, 5.1, 5.01, 5.001]

for t in intervals:
    avg = average_rate_of_change(distance, 5, t)
    print(f"Average speed from 5 to {t}: {avg}")

Average speed from 5 to 5.5: 51.45000000000002
Average speed from 5 to 5.1: 49.490000000000016
Average speed from 5 to 5.01: 49.04899999999903
Average speed from 5 to 5.001: 49.004899999997754


#d ans= As the speed gets smaller, the average speed is 49 m/s.

In [17]:
#e: find the deriative
import sympy as sp

# Define the variable
t = sp.symbols('t')

# Define the function f(t) = 4.9 * t^2
f = 4.9 * t**2

# Differentiate f with respect to t
f_prime = sp.diff(f, t)

print("f(t) =", f)
print("f'(t) =", f_prime)


f(t) = 4.9*t**2
f'(t) = 9.8*t


In [27]:
#f: Evaluate the deriative
instantaneous_speed = 9.8 * 5

print("Instantaneous speed at t = 5:", instantaneous_speed)

Instantaneous speed at t = 5: 49.0


#f written portion answer: The derivative gives an instantaneous speed of 49 m/s, which matches the value in part d. As the time interval becomes smaller, the average rate of change approaches the derivative, which represents the instantaneous rate of change.

In [28]:
#Calculating and Interpreting Partial Derivatives
def price(C, Y):
    return 16000 + 2400*C - 1800*Y

selling_price = price(8, 5)

print("Predicted Selling Price: $", selling_price)

Predicted Selling Price: $ 26200


In [29]:
#b Find and interpret value
partial_C = 2400

print("∂P/∂C =", partial_C)

∂P/∂C = 2400


#b interpretation: Each condition point adds $2,400 to the price, assuming the car's age is the same. Better condition means a higher price.

In [30]:
#c Find and interpret value 
partial_Y = -1800

print("∂P/∂Y =", partial_Y)

∂P/∂Y = -1800


#c interpretation: For each year the car ages, the predicted selling price decreases by $1,800, assuming the condition rating remains constant.